src/sources/channel.rs

```diff
@@ -85,19 +84,26 @@ impl<T> SyncSender<T> {
     /// deliver the message. Readiness is signalled to the event loop
     /// *before* blocking.
```

Comment:
I feel like you should remove the last sentence, since it's talking purely about 'internals', that are not meaningful to the user.

Reply:
This seems like internals, but it is observable behavior, similar to the issue you reported in the chatroom. This guarantees that the event loop will be woken up and thus that the app should not block indefinitely.

Reply:
>This seems like internals, but it is observable behavior, similar to the issue you reported in the chatroom. This guarantees that the event loop will be woken up and thus that the app should not block indefinitely.

But I'd assume that user expects that adding things to source should wake up a channel? In anyway, you should s/before/after, you really want to keep that.

Reply:
Well, no? If you block before signalling readiness, then the event loop will never wake up, the channel will never be emptied, and you'll never stop blocking... ?

Reply:
If we get `Disconnected`, that means that the receiver side of the channel has been dropped, so there is nothing to ping anymore.

Reply:
I mean that you can't wake up event loop if you've disconnected, so it's fine to ping, since it's 'free'?

Reply:
It one syscall, so it's not completely free. So yeah, calling the ping method would work, but there is no point in doing so.